src/include/sof/audio/module_adapter/module/generic.h

```diff
@@ -132,6 +132,8 @@ struct module_resources {
 	struct list_item cont_chunk_list;	/**< Memory container chunks */
 	size_t heap_usage;
 	size_t heap_high_water_mark;
+	struct k_heap *heap;
+	void *heap_mem;
```

Comment:
The `k_heap` structure embeds a `sys_heap`, which already contains fields for the heap's base address and memory size. These can be reused instead of introducing new redundant field.